src/System.Formats.Tar.TestData/GenerateTarFiles.sh

```diff
@@ -0,0 +1,472 @@
+{
+    yellow="\e[33m"
+    Echo $yellow "$1"
+}
+
+function EchoSuccess()
+{
+    green="\e[32m"
+    Echo $green "$1"
+}
+
+function EchoInfo()
+{
+    cyan="\e[36m"
+    Echo $cyan "$1"
+}
+
+function CheckLastErrorOrExit()
+{
+    message=$1
+
+    if [ $? -ne 0 ]; then
+        EchoError "Failure: $message"
+        EchoError "Script failed to finish."
+        exit 1
+    else
+        EchoSuccess "Success: $message"
+    fi
+}
+
+function ConfirmDirExists()
+{
+    Dir=$1
+
+    if [ ! -d $Dir ]; then
+        EchoError "Directory did not exist: $Dir"
+        exit 1
+    fi
+}
+
+function DeleteAndRecreateDir()
+{
+    Dir=$1
+
+    if [ -d $Dir ]; then
+        EchoWarning "Deleting folder: $Dir"
+        sudo rm -r $Dir
+    fi
+
+    EchoWarning "Creating folder: $Dir"
+    mkdir $Dir
+
+    ConfirmDirExists $Dir
+}
+
+function ExecuteTar()
+{
+    FullPathFolderToArchive=$1
+    Arguments=$2
+    FileName=$3
+    Format=$4
+    WithGEA=$5
+
+    EchoSuccess "----------------------------------------------"
+
+    GEAArgument=""
+    FormatArgument="$Format"
+    if [ $Format = "pax_gea" ] && [ $WithGEA = 1 ]; then
+        EchoWarning "Creating extra pax file with global a extended attributes entry"
+        FormatArgument="pax"
+        GEAArgument="$GEAPAXOptions"
+    fi
+
+    # IMPORTANT: This will ensure we archive entries that have relative paths to this folder
+    EchoInfo "cd $FullPathFolderToArchive"
+    cd $FullPathFolderToArchive
+
+    TarCommand="tar $Arguments $FileName * --format=$FormatArgument $GEAArgument"
+    EchoInfo "$TarCommand"
+
+    # Execute the command as the user that owns the files
+    # to archive, otherwise tar fails to pack them
+    sudo $TarCommand
+
+    if [ $? -ne 0 ]; then
+        EchoError "Tar command failed!"
+        if [ -f $FileName ]; then
+            EchoError "Deleting malformed file: $FileName"
+            sudo rm $FileName
+        fi
+    else
+        EchoSuccess "Tar archive created successfully: $FileName"
+    fi
+
+    EchoSuccess "----------------------------------------------"
+}
+
+function GenerateArchive()
+{
+    DirsRoot=$1
+    TargetDir=$2
+    Arguments=$3
+    Extension=$4
+
+    UnarchivedDir="$DirsRoot/unarchived"
+    FoldersToArchiveArray=($(ls $UnarchivedDir))
+
+    for Format in "${FormatsArray[@]}"; do
+
+        OutputDir="$TargetDir/$Format"
+        DeleteAndRecreateDir $OutputDir
+
+        for FolderToArchive in "${FoldersToArchiveArray[@]}"; do
+
+            FullPathFolderToArchive="$UnarchivedDir/$FolderToArchive/"
+            FileName="$OutputDir/$FolderToArchive$Extension"
+
+            WithGEA=0
+            if [ $Format = "pax_gea" ]; then
+                WithGEA=1
+            fi
+
+            ExecuteTar "$FullPathFolderToArchive" "$Arguments" "$FileName" "$Format" $WithGEA
+
+        done
+    done
+
+    # Tar was executed elevated, need to ensure the
+    # generated archives are readable by current user
+    ResetOwnership $TargetDir
```

Comment:
I added instructions indicating that the script needs to be executed as sudo.

Also, this script isn't expected to be executed often. But if it needs to be, then it would be very noticeable if something didn't go right by looking at the `git status`.